config/dynamic-scaling-configuration.yaml

```diff
@@ -3,10 +3,28 @@
 #The following properties must be defined for each instance types compute node configuration:
 #   - max_compute_nodes: the maximum number of nodes needed to create an appropriate machine pool with the configuration needed by the autoscaler
 
+# For each supported cloud provider, the following properties must be provided for the machine type
+#   - cluster_wide_workload_machine_type: the instance type size to handle cluster wide workloads e.g ingress controllers, observability operators ec 
+#   - kafka_workload_machine_type: the instance type size to handle the Kafka instance workload e.g brokers, kafka-admin API etc
+
 ---
-developer:
-  compute_nodes_config:
-    max_compute_nodes: 3
-standard:
-  compute_nodes_config:
-    max_compute_nodes: 9
+# The version of OpenShift on newly auto created data plane clusters.
+# If the value is empty (default), then the latest version is going to be used. 
+new_data_plane_openshift_version: ""
+# Whether to auto create a data plane  cluster 
+enable_dynamic_data_plane_scale_up: false
+# Whether to auto delete a data plane cluster cluster
+enable_dynamic_data_plane_scale_down: false
+# machine type configuration per cloud provider 
+machine_type_per_cloud_provider:
+  aws:
```

Comment:
We need to apply the same to gcp

Reply:
> A decision can't only be taken for aws

Correct but this change was based on the default supported providers in the file https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/55c1891abb04c33deea85233da03c08b917a2e01/config/provider-configuration.yaml which at the moment we've only `aws` enabled.

Reply:
We should change it now even if it's disabled so we get some defaults. Let's make sure we know the values for them

Reply:
The gcp machine types are not affected for the https://issues.redhat.com/browse/MGDSTRM-9802, the previous communicated values are ones we'll start with. 
As for putting the default values, I am -1/+1 on this. +1 on having the default, -1 on not having the default for something that's not used at all by default, plus the values can be easily added in the configuration.

Reply:
> As for putting the default values, I am -1/+1 on this. +1 on having the default, -1 on not having the default for something that's not used at all by default, plus the values can be easily added in the configuration.

I think we should put it. Even if you don't enable the provider by default it is prepared so when you enable the provider is already configured properly. Otherwise when enabling it you would need to add/modify N configurations in M configuation files

Reply:
> The gcp machine types are not affected for the change in https://issues.redhat.com/browse/MGDSTRM-9802, the previous communicated values are ones we'll start with.

Exactly, and that's my concern. I understand that evaluation should not be limited to AWS

Reply:
Okay make sense to avoid the configuration modification dance. Let me add gcp as well. I'll probably add it on the service template as well for the same reasoning.

Reply:
pushed the default value inclusion.

> Exactly, and that's my concern. I understand that evaluation should not be limited to AWS

Agree, it'll have been best to have the gcp machine type now, to avoid all those futures hustles for existing clusters. The understanding I have is, finding the most cost effective machine type for gcp will come at a later phase (you know this more than myself, so my understanding could be wrong). But for now, and for that work on https://issues.redhat.com/browse/MGDSTRM-9802, the focus is on AWS.

Reply:
My understanding at the time was that the most cost effective machine type coming at a later phase applied to both AWS and GCP instances.
Can we ping them in the issue to make sure they are considering that and about the current progress of it?

Reply:
This was talked about last Firday in the dynamic scaling sync call. I'll ping you on it.